README.md

```diff
@@ -101,7 +101,7 @@ jobs:
         key: ${{ runner.os }}-primes
 
     - name: Generate Prime Numbers
-      if: steps.cache-primes.outputs.cache-hit != 'true'
+      if: ${{ steps.cache-primes.outputs.cache-hit != 'true' }}
```

Comment:
Do you have any debug logs of when this wasn't working?

`${{ }}` shouldn't be necessary in an `if` step 

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/expressions#about-expressions